presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -226,4 +247,73 @@ void nextAttempt()
             }
         }
     }
+
+    public static final class RetryStats
+    {
+        private final CounterStat success = new CounterStat();
+        private final CounterStat finalFailure = new CounterStat();
+        private final TimeDistribution failureLatency = new TimeDistribution(MICROSECONDS);
+        private final DistributionStat successRetry = new DistributionStat();
+        private final TimeDistribution successLatency = new TimeDistribution(MICROSECONDS);
+
+        @Managed
+        @Nested
+        public CounterStat getSuccess()
+        {
+            return success;
+        }
+
+        @Managed
+        @Nested
+        public CounterStat getFinalFailure()
+        {
+            return finalFailure;
+        }
+
+        @Managed
+        @Nested
+        public TimeDistribution getFailureLatency()
+        {
+            return failureLatency;
+        }
+
+        @Managed
+        @Nested
+        public DistributionStat getSuccessRetry()
+        {
+            return successRetry;
+        }
+
+        @Managed
+        @Nested
+        public TimeDistribution getSuccessLatency()
+        {
+            return successLatency;
+        }
+
+        public void addSuccess()
+        {
+            success.update(1);
+        }
+
+        public void addFinalFailure()
+        {
+            finalFailure.update(1);
+        }
+
+        public void addFailureLatency(Duration duration)
+        {
+            failureLatency.add(duration.roundTo(NANOSECONDS));
+        }
+
+        public void addSuccessRetry(int count)
```

Comment:
I noticed that now you don't record the number of attempts before failure. Why?

Reply:
Initially, I thought this value will be either 1 or number of retry. But it seems like, the time period between retry is random so this value can be something in between. I will add it.